mora/service/itsystem.py

```diff
@@ -182,3 +187,65 @@ def convert(start, end, effect):
             key=(lambda v: util.parsedatetime(v['valid_from']))
         ),
     )
+
+
+def validate_it(func):
+    @functools.wraps(func)
+    def wrapper(employee_uuid, req):
+        c = lora.Connector(virkningfra='-infinity', virkningtil='infinity')
+
+        errors = []
+
+        # TODO: cache anything from LoRA and reuse it in the actual request?
+
+        if not req.get(ITSYSTEM):
+            errors.append('missing "itsystem"')
+        else:
+            if not util.is_uuid(req[ITSYSTEM].get('uuid')):
+                errors.append('missing or invalid "itsystem" UUID')
+
+            elif not c.itsystem.get(uuid=req[ITSYSTEM]['uuid']):
+                errors.append('no such it system')
+
+        if not req.get('valid_from'):
+            errors.append('missing or invalid start date')
+
+        if not c.bruger.get(uuid=employee_uuid):
+            errors.append('no such user')
+
+        # TODO: ensure effective time is within both user and itsystem
+
+        if errors:
+            # TODO: add granular, consistent and documented error reporting
+            raise ValueError('; '.join(errors))
+
+        return func(employee_uuid, req)
+
+    return wrapper
+
+
+@validate_it
+def create_system(employee_uuid, req):
+    systemid = req[ITSYSTEM].get('uuid')
+    valid_from = req.get('valid_from')
+    valid_to = req.get('valid_to') or 'infinity'
```

Comment:
There's constants for valid_from and valid_to in keys.py - just for consistency.